Documentation/building/build-guidelines.md

```diff
@@ -0,0 +1,47 @@
+Build guidelines
+================
+###Goal
+
+It is difficult for devs, who are new to the corefx repo, to reason about the build and understand how to implement new features, or otherwise update the repo's build.  The primary goal is to provide a repo which is easier to reason about.  A secondary goal, would be to help the dev implement their changes to the build (without getting in their way) such that they are confident they are following our guidelines.
+
+- Keep it simple
+- Keep it clean
+- Make it maintainable (secondary)
+
+### Keep it simple
+
+We don't want multiple ways to do the same thing.  There is a support cost and mind-share cost associated with having multiple ways to achieve something.  It's ok to have insider tricks, things for power-users who understand the default experience but want to optimize it for their development patterns.
```

Comment:
I agree with the general principle but I'd be interested in what you had in mind as I suspect there will always be more then one way to do things (i.e. msbuild, build.cmd, run.cmd, VS). However as much as possible we should ensure each of those are just an alias for the core build call.